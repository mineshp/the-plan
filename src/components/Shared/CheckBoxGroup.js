import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox, Form } from 'semantic-ui-react';

const CheckBoxGroup = ({
    data,
    assigned,
    handleCheckBoxChange
}) => {
    const CheckBoxItems = [];
    data.map((item) => {
        const keyId = item._id; // eslint-disable-line no-underscore-dangle
        const isChecked = assigned.includes(item.name);
        const checkedState = isChecked ? 'true' : 'false';

        return CheckBoxItems.push(
            <Form.Field key={keyId}>
                <Checkbox
                    label={item.name}
                    id={keyId}
                    data-cb={item.name}
                    onChange={handleCheckBoxChange}
                    checked={isChecked}
                    value={checkedState}
                />
            </Form.Field>
        );
    });
    return (
        <div>{ CheckBoxItems }</div>
    );
};

CheckBoxGroup.propTypes = {
    handleCheckBoxChange: PropTypes.func.isRequired,
    data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
    assigned: PropTypes.arrayOf(PropTypes.string)
};

CheckBoxGroup.defaultProps = {
    assigned: null
};

export default CheckBoxGroup;

