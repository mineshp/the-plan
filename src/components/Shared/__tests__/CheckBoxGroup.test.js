import React from 'react';
import renderer from 'react-test-renderer';
import CheckBoxGroup from '../CheckBoxGroup';
import { mockListProfiles } from '../../../helpers/test/testData/controlCentreData';

const mockProfiles = mockListProfiles();

it('renders correctly for an item that is not set to checked as default', () => {
    const handleCheckBoxChange = jest.fn();

    const tree = renderer.create(<CheckBoxGroup
        data={mockProfiles}
        handleCheckBoxChange={handleCheckBoxChange}
        assigned={[]}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly for an item that is set to checked as default', () => {
    const handleCheckBoxChange = jest.fn();

    const tree = renderer.create(<CheckBoxGroup
        data={mockProfiles}
        handleCheckBoxChange={handleCheckBoxChange}
        assigned={['profileX', 'HOME_DIY']}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});
