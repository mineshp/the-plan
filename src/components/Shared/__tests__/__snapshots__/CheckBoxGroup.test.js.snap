// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for an item that is not set to checked as default 1`] = `
<div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      data-cb="MIN_JIGS_WEDDING"
      id="1234567"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value="false"
      />
      <label>
        MIN_JIGS_WEDDING
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      data-cb="HOME_DIY"
      id="1234568"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value="false"
      />
      <label>
        HOME_DIY
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      data-cb="CHRISTMAS"
      id="1234569"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value="false"
      />
      <label>
        CHRISTMAS
      </label>
    </div>
  </div>
</div>
`;

exports[`renders correctly for an item that is set to checked as default 1`] = `
<div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      data-cb="MIN_JIGS_WEDDING"
      id="1234567"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value="false"
      />
      <label>
        MIN_JIGS_WEDDING
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checked checkbox"
      data-cb="HOME_DIY"
      id="1234568"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={true}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value="true"
      />
      <label>
        HOME_DIY
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      data-cb="CHRISTMAS"
      id="1234569"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name={undefined}
        readOnly={true}
        tabIndex={0}
        type="checkbox"
        value="false"
      />
      <label>
        CHRISTMAS
      </label>
    </div>
  </div>
</div>
`;
